streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -228,12 +226,17 @@ public Subscription subscription(Set<String> topics) {
         // 2. within each client, tasks are assigned to consumer clients in round-robin manner.
         Map<UUID, Set<String>> consumersByClient = new HashMap<>();
         Map<UUID, ClientState<TaskId>> states = new HashMap<>();
-
+        SubscriptionUpdates subscriptionUpdates = streamThread.builder.getSubscriptionUpdates();
```

Comment:
Instead of exposing the `builder.getSubscriptionUpdates` in `StreamPartitionAssignor`, could we just add a function named `updateSubscriptions` in `TopologyBuilder`?